src/main/java/com/aws/greengrass/mqtt/bridge/model/Message.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.mqtt.bridge.model;
+
+/**
+ * Common representation of a Message.
+ */
+public interface Message {
+    Message toPubSub();
+
+    Message toMqtt();
+
+    String getTopic();
+
+    byte[] getPayload();
+
+    Message ofTopic(String topic);
```

Comment:
why does this exist? We already have a `Message`, so why are we then calling a method on it to give us a different message which a presumably different topic?

Reply:
Yeah, cause of generics weirdness https://github.com/aws-greengrass/aws-greengrass-mqtt-bridge/pull/88/files#diff-7d623726e13974aaebfada082983f1677688fcc2f9225dc9a51c63a580ee89c9R145-R150.  I guess could just make it a setTopic, but that would remove the immutability of message

Reply:
I think the name is confusing. How about `newFromMessageWithTopic`?

Reply:
yeah will change

Reply:
done